models/nc_news.models.js

```diff
@@ -34,18 +34,43 @@ function fetchArticleById(article_id) {
 function fetchCommentsByArticleId(article_id) {
   return db
     .query(
-      `SELECT articles.article_id, comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body FROM articles
-      JOIN comments
-      ON articles.article_id = comments.article_id
-      WHERE articles.article_id = $1
-      ORDER BY comments.created_at DESC;`,
+      `SELECT * FROM comments 
+      WHERE article_id = $1 
+      ORDER BY created_at DESC;`,
       [article_id]
     )
     .then(({ rows }) => {
-      if (rows.length === 0) {
+      return rows;
+    });
+}
+
+function checkIfArticleExists(article_id) {
+  return db
+    .query(
+      `SELECT * FROM articles 
+      WHERE article_id = $1`,
+      [article_id]
+    )
+    .then(({ rows: articles }) => {
+      if (articles.length === 0) {
         return Promise.reject({ status: 404, message: "Article not found." });
       }
-      return rows;
+    });
+}
+
+function addCommentByArticleId(article_id, { username, body }) {
+  const author = username;
+  const votes = 0;
```

Comment:
you could have a promise.reject here that if there was no username or body it wouldnt go on